File: 백준/Silver/8911. 거북이/거북이.py
import sys as s

def turtle(command):
    x, y = 0, 0  # 현재 위치
    direction = 0  # # 0: 북, 1: 동, 2: 남, 3: 서 (리스트 위치별 방향)

    dx = [0, 1, 0, -1]  # 북 : 0 동 : 1 남 : 0 서 : -1
    dy = [1, 0, -1, 0]

    visited = [(x, y)]

    for c in command:
        if c == 'F':
            x += dx[direction]
            y += dy[direction]
        elif c == 'B':
            x -= dx[direction]
            y -= dy[direction]
        elif c == 'L':
            direction = (direction + 3) % 4
        elif c == 'R':
            direction = (direction + 1) % 4
        visited.append((x, y))

    # •	direction = (direction + 3) % 4: 왼쪽으로 90도 회전
    # •	현재 방향에서 3을 더하고 4로 나눈 나머지를 계산합니다. 이는 시계 반대 방향으로 90도 회전한 효과를 줍니다.
    # •	예: 방향이 북쪽(0)일 때, (0 + 3) % 4 = 3이므로 서쪽(3)이 됩니다.
    # •	direction = (direction + 1) % 4: 오른쪽으로 90도 회전
    # •	현재 방향에서 1을 더하고 4로 나눈 나머지를 계산합니다. 이는 시계 방향으로 90도 회전한 효과를 줍니다.
    # •	예: 방향이 북쪽(0)일 때, (0 + 1) % 4 = 1이므로 동쪽(1)이 됩니다.

    # x 좌표의 최소값을 구합니다.
    min_x = min(v[0] for v in visited)  # v[0]은 각 튜플의 첫 번째 요소, 즉 x 값을 의미합니다.
    # x 좌표의 최대값을 구합니다.
    max_x = max(v[0] for v in visited)  # v[0]은 각 튜플의 첫 번째 요소, 즉 x 값을 의미합니다.
    # y 좌표의 최소값을 구합니다.
    min_y = min(v[1] for v in visited)  # v[1]은 각 튜플의 두 번째 요소, 즉 y 값을 의미합니다.
    # y 좌표의 최대값을 구합니다.
    max_y = max(v[1] for v in visited)  # v[1]은 각 튜플의 두 번째 요소, 즉 y 값을 의미합니다.

    area = (max_x - min_x) * (max_y - min_y)
    return area

n = int(s.stdin.readline().strip())
command = [s.stdin.readline().strip() for _ in range(n)]
result = []

for i in command:
    result.append(str(turtle(i)))

print('\n'.join(result))

File: 백준/Silver/8911. 거북이/거북이.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.ArrayList;
import java.util.List;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int n = Integer.parseInt(br.readLine().trim());
        
        String[] commands = new String[n];
        for (int i = 0; i < n; i++) {
            commands[i] = br.readLine().trim();
        }
        
        StringBuilder sb = new StringBuilder();
        for (String command : commands) {
            sb.append(turtle(command)).append('\n');
        }
        
        System.out.print(sb.toString());
    }
    
    static int turtle(String command) {
        int x = 0, y = 0; 
        int direction = 0;  

        int[] dx = {0, 1, 0, -1};  
        int[] dy = {1, 0, -1, 0};  

        List<int[]> visited = new ArrayList<>();
        visited.add(new int[]{x, y});

        for (char c : command.toCharArray()) {
            if (c == 'F') {
                x += dx[direction];
                y += dy[direction];
            } else if (c == 'B') {
                x -= dx[direction];
                y -= dy[direction];
            } else if (c == 'L') {
                direction = (direction + 3) % 4;  
            } else if (c == 'R') {
                direction = (direction + 1) % 4;  
            }
            visited.add(new int[]{x, y});
        }

        int minX = Integer.MAX_VALUE;
        int maxX = Integer.MIN_VALUE;
        int minY = Integer.MAX_VALUE;
        int maxY = Integer.MIN_VALUE;
        
        for (int[] pos : visited) {
            minX = Math.min(minX, pos[0]);
            maxX = Math.max(maxX, pos[0]);
            minY = Math.min(minY, pos[1]);
            maxY = Math.max(maxY, pos[1]);
        }

        return (maxX - minX) * (maxY - minY);
    }
}